### ISSUE TITLE
System builder not obeying 'paths' set in config (0.12.2)

### ISSUE THREAD

**Issue by kenkouot** [systemjs/builder#219]
When using SystemJS in the browser with ES6 modules, it knows to map the following import statements correctly:
```js
// This will get mapped to <example expanded path>/path/to/foo.js
import foo from './path/to/foo';
```
However, when running the same config on `0.12.2` of the builder, I am consistently getting an error that looks like this: `Error: ENOENT, open '<example expanded path>/path/to/foo']`. It seems to not be resolving the `*.js`.

I've inspected my `builder.loader.paths` and it looks fine:
```js
  paths:
   { traceur: 'file:///<redacted>/node_modules/systemjs-builder/node_modules/traceur/bin/traceur.js',
     'traceur-runtime': 'file:///<redacted>/node_modules/systemjs-builder/node_modules/traceur/bin/traceur-runtime.js',
     '*': '*.js',
     'github:*': 'jspm_packages/github/*.js',
     'npm:*': 'jspm_packages/npm/*.js' },
```

**Comment by guybedford** [comment]
Make sure you run the right versions of SystemJS and builder together. SystemJS 0.16 goes with SystemJS builder 0.11. SystemJS 0.17/0.18 goes with SystemJS builder 0.12. See the SystemJS release notes at https://github.com/systemjs/systemjs/releases/tag/0.17.0 for more information on the changes.

**Comment by nathanboktae** [comment]
`systemjs` is a direct dependency of `systemjs-builder` and is declared as `^0.18.0`. It's not possible for the end user to mess this up.

I am seeing the exact same issue, with the systemjs direct dependency resolved as `0.18.4`. Please reopen and visit this, thanks.

**Comment by guybedford** [comment]
@nathanboktae good point, ok it seems the issue is that adding '.js' extensions via paths rules like `'*': '*.js'` is not supported in SystemJS 0.18 and builder 0.12. Rather use the `defaultJSExtensions` option, or `defaultExtension` package configuration option. I've updated the release notes at https://github.com/systemjs/systemjs/releases/tag/0.17.0 to reflect this, it should have been mentioned there.
